client/ayon_resolve/plugins/publish/precollect_shots.py

```diff
@@ -0,0 +1,86 @@
+import pyblish
+
+from ayon_resolve.otio import utils
+
+
+class PrecollectShot(pyblish.api.InstancePlugin):
+    """PreCollect new shots."""
+
+    order = pyblish.api.CollectorOrder - 0.48
+    label = "Precollect Shots"
+    hosts = ["resolve"]
+    families = ["shot"]
+
+    @staticmethod
+    def _prepare_context_hierarchy(instance):
+        """
+        TODO: explain
+        resolve:
+        https://github.com/ynput/ayon-core/blob/6a07de6eb904c139f6d346fd6f2a7d5042274c71/client/ayon_core/plugins/publish/collect_hierarchy.py#L65
+
+        traypublisher:
+        https://github.com/ynput/ayon-traypublisher/blob/develop/client/ayon_traypublisher/plugins/publish/collect_shot_instances.py#L188
+        """
+        instance.data["folderPath"] = instance.data.pop("hierarchy_path")
+        instance.data["integrate"] = False  # no representation for shot
```

Comment:
This is more of a work-around for now, but probably something worth to discuss with @jakubjezek001 and  @iLLiCiTiT.
* `folderPath` as to be set to an existing folder for the Creator to detect the instance has valid context
* `folderPath` has to be the hierarchy of the shot to create for "Collect Hierarchy" plugin to properly [compute the final path](  https://github.com/ynput/ayon-core/blob/6a07de6eb904c139f6d346fd6f2a7d5042274c71/client/ayon_core/plugins/publish/collect_hierarchy.py#L65)

As a work-around this code:
1. Create a instance with `folderPath` containing the selected parent context folder at Creation to produce a valid instance
2. Immediately replace `folderPath` with final hierarchy to create in early collect plugins

Reply:
Once `newHierarchyIntegration` will be included you can fill `folderPath` value with Ayon marker value `folder_path`

Reply:
I guess there is something I'm doing wrong then.
I have set `newHierarchyIntegration = True` to each of the created instances.

Here is what I get if I set `folderPath` to Ayon marker value `folder_path` when creating the instance:
![invalid_context](https://github.com/user-attachments/assets/bd86549a-0e93-4d37-94a5-a175eede5800)

This is what I get with my hack
(set `folderPath` to parent context when creating it then adjust it at precollect time):
![valid_context](https://github.com/user-attachments/assets/eabad5a7-6170-4901-9606-74369d76f66f)

Reply:
Yeah you are quite right, the creating context is not supporting this attribute. So the hack is actually the same workaround as it is done in TP. @iLLiCiTiT question is if we could improve the workflow now so we do not need to hack this?

Reply:
Not sure when this PR should be merged? Not sure if I could get to it in next 2 weeks.

Reply:
This PR is adjusted to report changes from https://github.com/ynput/ayon-core/pull/894/files